vnext/JSI/Shared/ChakraObjectRef.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "jsi/jsi.h"
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
```

Comment:
(nit) isn't this a noop?

Reply:
I am not sure what you mean here. Before #including jsrt.h, we need to #define USE_EDGEMODE_JSRT if it has not been #defined already. Do I have a typo that I am not seeing?

---
In reply to: [336143472](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336143472) [](ancestors = 336143472)

Reply:
If instead of:
#ifndef USE_EDGEMODE_JSRT
#define USE_EDGEMODE_JSRT
#endif

You will have:
#define USE_EDGEMODE_JSRT

---
In reply to: [336153524](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336153524) [](ancestors = 336153524,336143472)

Reply:
We cannot remove those lines. If we do, and USE_EDGEMODE_JSRT is already defined in a different header or through the vcxproj file, we get a compile error for macro redefinition.

---
In reply to: [336180453](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336180453) [](ancestors = 336180453,336153524,336143472)

Reply:
Got it, thanks for checking.

---
In reply to: [336702860](https://github.com/microsoft/react-native-windows/pull/3445#discussion_r336702860) [](ancestors = 336702860,336180453,336153524,336143472)